### PR TITLE
[DPM-745] generate referral id

### PR DESCRIPTION
**Pull request by georgiypetrov**

### CODE REVIEW

File: referrals/usecase/usecase.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+func (r *ReferralsUseCase) GetOrCreateReferralID(ctx context.Context, accountName string) (string, error) {
+	hasRefID, err := r.repo.HasReferralID(ctx, accountName)
+	if err != nil {
+		log.Debug().Msgf("Referral ID exist check error: %s", err.Error())
+		return "", err
+	}
+
+	if !hasRefID {
+		refID, err := referrals.GenerateRandomString(ReferralIDLen)
```
Comment (justefg) [inline referrals/usecase/usecase.go:27]: why wouldn't empty ref work?

File: migrations/0011_referrals_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE referrals
+(
+    account_name VARCHAR(13) REFERENCES users(account_name),
+    referral_id VARCHAR NOT NULL
+);
+
+CREATE INDEX referral_id_idx ON referrals (referral_id);
+
```
Comment (justefg) [inline migrations/0011_referrals_table.up.sql:8]: should we create index on account_name as well to optimize getting a referral for a user?

File: referrals/repository.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package referrals
+
+import "context"
+
+type Repository interface {
+	HasReferralID(ctx context.Context, accountName string) (bool, error)
+	GetReferralID(ctx context.Context, accountName string) (string, error)
```
Comment (justefg) [inline referrals/repository.go:7]: can we remove `HasReferralID` and only have `GetReferralID` which return "" if the user doesn't have a referral?

Comment (georgiypetrov) [reply referrals/repository.go:7]: ok

File: migrations/0011_referrals_table.up.sql (hunk 2 above)
Comment (georgiypetrov) [reply migrations/0011_referrals_table.up.sql:8]: should we? primary key is unique by default

Comment (georgiypetrov) [reply migrations/0011_referrals_table.up.sql:8]: yes, we should, it will be primary key because it's one-to-one relation

**Review by justefg — APPROVED** [review]

**Pull request merged**